### PR TITLE
r?fzzzy Fix selection

### PR DESCRIPTION
**Pull request by ianb**
Major refactoring and performance improvement to the selection

### CODE REVIEW

**Comment by ianb** [comment]
(I deployed this branch, because I wanted to get the homepage and timeout bug fixes out, but the branch still isn't actually merged)

File: addon/data/extractor-worker.js [hunk 1]
```diff
@@ -15,6 +15,35 @@ var FILENAME = "extractor-worker.js";
     */
 function extractData() {
   let start = Date.now();
+  // FIXME: need to include found images too
+  let startImage = Date.now();
```
Comment (fzzzy) [inline addon/data/extractor-worker.js]: This isn't needed any more, just use start instead

File: static/homepage/index.html [hunk 2]
```diff
@@ -133,9 +119,13 @@
             <h1 class="hero-header-text large">Share anything on the web.</h1>
             <h4 class="hero-tagline">Save, highlight, send
             </h4>
-            <p>
+            <p class="if-firefox">
```
Comment (fzzzy) [inline static/homepage/index.html]: I don't see how the globals in the js get defined to css classes

**Comment by fzzzy** [comment]
Ok, looks good with those nits

File: static/homepage/index.html (hunk 2 above)
Comment (ianb) [inline static/homepage/index.html]: The JS adds a class to body, and the CSS uses selectors to hide or show these classes based on the body class.  I put a comment into the page JS to explain.

**Pull request merged**